File: 3005. Count Elements With Maximum Frequency.c
/*
You are given an array nums consisting of positive integers.

Return the total frequencies of elements in nums such that those elements all have the maximum frequency.

The frequency of an element is the number of occurrences of that element in the array.
*/
int maxFrequencyElements(int* nums, int numsSize) {
    int cache[101];
    memset(cache, 0, sizeof(int)*101);
    for(int i = 0; i < numsSize; i++){
        cache[nums[i]]++;
    }
    int largest = cache[0];
    for(int i = 1; i < 101; i++){
        largest = largest < cache[i] ? cache[i] : largest;
    }
    int count = 0;
    for(int i = 1; i < 101; i++){
        count = largest == cache[i] ? count+1 : count;
    }
    return count*largest;
}

File: 3005. Count Elements With Maximum Frequency.py
'''
You are given an array nums consisting of positive integers.

Return the total frequencies of elements in nums such that those elements all have the maximum frequency.

The frequency of an element is the number of occurrences of that element in the array.
'''
class Solution(object):
    def maxFrequencyElements(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        d = {}
        for num in nums:
            if num in d:
                d[num] += 1
            else:
                d[num] = 1
        maxFrequency = max([d[k] for k in d.keys()])
        return sum([d[k] for k in d.keys() if d[k] == maxFrequency])